src/modules/local_position_estimator/BlockLocalPositionEstimator.cpp

```diff
@@ -196,37 +206,43 @@ void BlockLocalPositionEstimator::update()
 	float dt = (newTimeStamp - _timeStamp) / 1.0e6f;
 	_timeStamp = newTimeStamp;
 
-	//printf("dt: %0.5g\n", double(dt));
-
 	// set dt for all child blocks
 	setDt(dt);
 
-	// see which updates are available
-	bool flowUpdated = _sub_flow.updated();
-	bool paramsUpdated = _sub_param_update.updated();
-	bool baroUpdated = _sub_sensor.updated();
-	bool lidarUpdated = false;
-	bool sonarUpdated = false;
-
-	if (_sub_distance.updated()) {
-		if (_sub_distance.get().type == distance_sensor_s::MAV_DISTANCE_SENSOR_LASER) {
-			lidarUpdated = true;
-		}
+	// auto-detect connected rangefinders while not armed
+	if (!_sub_armed.get().armed && (_sub_lidar == NULL || _sub_sonar == NULL)) {
+		for (int i = 0; i < ORB_MULTI_MAX_INSTANCES; i++) {
+			if (_distance_subs[i]->get().timestamp == 0) {
+				continue; // ignore sensors with no data coming in
+			}
 
-		if (_sub_distance.get().type == distance_sensor_s::MAV_DISTANCE_SENSOR_ULTRASOUND) {
-			sonarUpdated = true;
-		}
+			if (_distance_subs[i]->get().type == distance_sensor_s::MAV_DISTANCE_SENSOR_LASER &&
+			    _sub_lidar == NULL) {
+				_sub_lidar = _distance_subs[i];
+				warnx("[lpe] Lidar detected with ID %i", i);
 
-		if (_sub_distance.get().type == distance_sensor_s::MAV_DISTANCE_SENSOR_INFRARED) {
-			mavlink_log_info(_mavlink_fd, "[lpe] no support to short-range infrared sensors ");
-			warnx("[lpe] short-range infrared detected. Ignored... ");
+			} else if (_distance_subs[i]->get().type == distance_sensor_s::MAV_DISTANCE_SENSOR_ULTRASOUND &&
+				   _sub_sonar == NULL) {
+				_sub_sonar = _distance_subs[i];
+				warnx("[lpe] Sonar detected with ID %i", i);
+			}
 		}
 	}
 
+	// see which updates are available
+	bool flowUpdated = _sub_flow.updated();
+	bool paramsUpdated = _sub_param_update.updated();
+	bool baroUpdated = _sub_sensor.updated();
 	bool gpsUpdated = _sub_gps.updated();
 	bool homeUpdated = _sub_home.updated();
 	bool visionUpdated = _sub_vision_pos.updated();
 	bool mocapUpdated = _sub_mocap.updated();
+	bool lidarUpdated = false;
+	bool sonarUpdated = false;
+
```

Comment:
So did we have some issue with lidar enabling in flight? And this we why we init it on the ground first?